cpp/roaring.hh

```diff
@@ -45,6 +45,18 @@ class Roaring {
         }
         roaring.copy_on_write = r.roaring.copy_on_write;
     }
+    /**
+     * Move constructor
+     */
+    Roaring(Roaring &&r) {
+        bool is_ok = ra_init(&roaring.high_low_container);
+        if (!is_ok) {
+            throw std::runtime_error("failed memory alloc in constructor");
+        }
+        roaring.copy_on_write = false;
+        
+        std::swap(roaring, r.roaring);
```

Comment:
Move semantic requires to leave the right-side argument in a valid state. The desired behaviour is:

```
Roaring foo;
foo.add(1);

Roaring bar(std::move(foo)); // bar got data from foo, but
foo.add(2); // <<< this still must be possible
```

If we used merely std::move as you wrote, it would transfer all pointers from the underlaying roaring_bitmap_t structure. Transfer of bare pointers means setting them NULL on the right side. And then the last line in the above example would cause segfault.